### DISCUSSION TITLE
Metadata for entity X not found

### DISCUSSION THREAD

**Discussion by loss-and-quick** [mikro-orm/mikro-orm#4367] [Q&A] [answered]
`orm.em.getRepository(Worker).findAll()` - cause this error

**Stack trace**
```
        return new MetadataError(`Metadata for entity ${entity} not found`);
               ^

MetadataError: Metadata for entity Worker not found
    at MetadataError.missingMetadata (node_modules/@mikro-orm/core/errors.js:178:16)
    at MetadataStorage.get (node_modules/@mikro-orm/core/metadata/MetadataStorage.js:54:42)
    at SqlEntityManager.getRepository (node_modules/@mikro-orm/core/EntityManager.js:65:40)
    at SqlEntityManager.getRepository (node_modules/@mikro-orm/knex/SqlEntityManager.js:41:22)
    at DatabaseService.getWorkers (Database/Database.sevice.js:23:28)
    at App.<anonymous> (index.js:19:40)
    at Generator.next (<anonymous>)
    at fulfilled (index.js:4:58) {
  entity: undefined
}
```

**To Reproduce**
Copy code from `Additional context`

**Expected behavior**
Should give me all workers

**Additional context**
``` ts
// MikroORM config
export default defineConfig({
    dbName: process.env.MIKRO_ORM_DB_NAME ?? 'db.sqlite',
    allowGlobalContext: true,
    persistOnCreate: true,
    debug: process.env.APP_ENV === 'development',
    entities: [Worker, JoinedGroup],
    tsNode: process.env.APP_ENV === 'development',
    forceEntityConstructor: true,
    resultCache: {
        expiration: 1000,
    },
});
```
``` ts
@Entity({ tableName: 'workers' })
export class Worker {
    @PrimaryKey({ trackChanges: false })
    readonly ID: number;

    // relations
    @OneToMany(() => JoinedGroup, (group) => group.worker)
    groups = new Collection<JoinedGroup>(this);

    @ManyToOne(() => Worker, { nullable: true })
    refferal?: Worker;

    @OneToMany(() => Worker, (worker) => worker.refferal)
    refferals = new Collection<Worker>(this);

    constructor(
        ID: number,
        refferal?: Worker,
    ) {
        this.ID = ID;
        this.refferal = refferal;
    }
}
```

**Versions**

| Dependency | Version |
| - | - |
| node | 18.15.0 |
| typescript | 5.0.4  |
| mikro-orm | 5.7.7 |
| sqlite |5.7.7 |

**Comment by B4nan** [comment]
This is a validation - it means you need to fix something in your code, not that there is a bug in the ORM.

I can't help more without seeing a complete repro for those things. Note that its the `em.getRepository` call that throws, not the `findAll`.

**Reply by B4nan** [reply]
What do you get when you call `em.getMetadata().getAll()`?

**Reply by loss-and-quick** [reply]
https://github.com/loss-and-quick/issue

**Reply by loss-and-quick** [reply]
> This is a validation - it means you need to fix something in your code, not that there is a bug in the ORM.

I thought it is bug or something

**Reply by B4nan** [reply, marked as answer]
You are discovering wrong `Worker` class:

https://github.com/loss-and-quick/issue/blob/main/src/configs/mikro-orm.config.ts#L2

**Reply by loss-and-quick** [reply]
Oh,sorry i didn't see this

**Reply by B4nan** [reply]
And I can see you have some markdown in that Worker entity file, so its not a valid TS and will fail the compilation probably

https://github.com/loss-and-quick/issue/blob/main/src/Database/Models/Worker.model.ts#L110

**Reply by loss-and-quick** [reply]
> And I can see you have some markdown in that Worker entity file, so its not a valid TS and will fail the compilation probably
> 
> https://github.com/loss-and-quick/issue/blob/main/src/Database/Models/Worker.model.ts#L110

I used it to paste into this issue
